### DISCUSSION TITLE
Blogging Challenge

### DISCUSSION THREAD

**Discussion by BekahHW** [Virtual-Coffee/virtualcoffee.io#711] [Monthly Challenge]
We'd love to have a collection of topics to have folks choose from for our November blogging challenge. If you have some ideas, please share the Type and the prompt.

## Examples
- **Testing** - Getting Started with Cypress
- **DevRel** - How to plan for a year of DevRel
- **Frontend** - How to Migrate a React Project to TypeScript

**Comment by jdwilkin4** [comment]
Here are my ideas 😄 

## HTML and CSS topics

- 10 HTML tags you never heard of or used
- How to use the new @when and @else container queries in CSS
- What I learned from building a project with X CSS library or framework
- What I learned from taking X HTML or CSS course
- How to build X using HTML & CSS
- What are some good courses for HTML and CSS
- Introduction to X HTML or CSS library
- Good open source project to help you practice HTML & CSS

## JavaScript

- Common JavaScript errors and how to solve them
- What I learned from building a project using X JavaScript library/framework
- How to build X using X JavaScript library/framework
- What I learned from taking X JavaScript course
- What are some good courses for JavaScript
- Introduction to X JavaScript library/framework
- Good open source project to help you practice JavaScript
- Good JS conferences or meetups to attend
- How to get started working with Node and Express
- Top X JavaScript books to read
- How to get started with Jest and React testing library
- Common TypeScript errors and how to fix them

## Accessibility

- What I learned from doing an accessibility audit of my website
- Good courses to learn about accessibility
- X common accessibility mistakes made on websites

## Hacktoberfest 

- What I learned from participating in my first Hacktoberfest
- What I learned from maintaining a repository during  Hacktoberfest
- What I learned from mentoring others during Hacktoberfest
- Good open source repositories I discovered during Hacktoberfest
- Overall review for Hacktoberfest 2022

## Technical writing

- How to get started with technical writing
- Best platforms for tech blogs
- How to write good headlines for technical blogs
- Tips for promoting a technical blog

## Testing

- Good online courses for testing

## General

- How to get started with GitHub actions
- Beginner guide to getting started with Git
- Advanced Git commands and concepts
- How to get started with X database
- Best tech podcasts to listen to

## Career

- How to land your first tech job
- How to use LinkedIn to connect with recruiters and hiring managers
- How to write a resume and cover letter without job experience
- How to network and build connections
- How to excel as a junior developer
- What I learned during my first 6 months on the job as a junior developer
- How to get promoted to mid level engineer
- How to become a senior developer
- What does a frontend engineer do?
- What does a backend engineer do?
- What does a DevOps engineer do?
- How to select a good bootcamp to start learning how to code
- Good online programs to learn how to code
- X common mistakes people make on their resumes
- What I learned from teaching myself how to code and landing a dev job
- Other higher paying tech jobs you might have not have heard of
- How to ace the tech interview
- How to ace the take home project
- How to deal with imposter syndrome as a developer
- What I have learned so far from X years working as a developer
- How to become an engineering manager
- How to negotiate for a better salary

That's all I got 😄

**Comment by Cerchie** [comment]
## Apache Kafka Topics

- What Is Apache Kafka
- What Are Producers and Consumers in Apache Kafka 
- What Are Apache Kafka Connectors and How To Use Them
- What Are The Differences Between Stream Processing and Batch Processing
- What Are The Types of Consumer Rebalancing 
- How To Write A Producer And Consumer Using Kafka.js 

## DevRel

- What I Learned From X Years in DevRel
- How To Plan For Running a Workshop
- The Role of User Metrics In DevRel
- How To Prepare a Presentation
- How To Start Connecting With a New Community
- How To Maintain a Genuine Twitter Presence

## JavaScript

- How To Pick A Web Dev Framework
- What Is Node.js, What Is Bun, and How Do I Pick Between Them 
- What Learning a Second Programming Language Taught Me About JS
- How to Do Math-y Things in Python vs. JS
- How To Write an SDK In JS

**Comment by ClJarvis** [comment]
This is a resource for the blogging challenge. 

Some tactics for writing in public 

https://jvns.ca/blog/2023/08/07/tactics-for-writing-in-public/
